testing/test_skipping.py

```diff
@@ -713,6 +714,27 @@ def doskip():
     assert result.ret == 0
 
 
+def test_skipped_folding(testdir):
+    testdir.makepyfile(
```

Comment:
i suspect this cant be solved without solving the mark ruin first - as such this pr is an enhancement and nice to have